### PR TITLE
[DOC] Add documentation for setting montage order

### LINKED ISSUE #5619: Adding a reference does not work after a montage has been set
The order in which you perform assigning a montage and adding a reference matters, which probably shouldn't be the case.

Consider the following example using the testing data from `mne/io/brainvision/tests/data/test.*` (the first two lines could be combined by supplying the `montage` argument in the call to `mne.io.read_raw_brainvision`):

```
raw = mne.io.read_raw_brainvision("test.vhdr", preload=True)
raw.set_montage("standard_1020")
raw = mne.io.add_reference_channels(raw, "T3")
```

This assigns the montage first and then adds a new reference channel which wasn't part of the data. This produces a `RuntimeWarning: The locations of multiple reference channels are ignored (set to zero).`, which results in an incorrect location of the newly added channel T3.

If instead you first add the reference channel and then set the montage, everything works as expected:

 ```
raw = mne.io.read_raw_brainvision("test.vhdr", preload=True)
raw = mne.io.add_reference_channels(raw, "T3")
raw.set_montage("standard_1020")
```

### PR DESCRIPTION
**Pull request by withmywoessner**
#### Reference issue
Example: Fixes #5619

#### What does this implement/fix?
Adds documentation that explains that the montage must be set after adding a reference.

#### Additional information
Should I also add a warning that runs inside the code itself after running `set_montage`?

### CODE REVIEW

**Comment by larsoner** [comment]
> Should I also add a warning that runs inside the code itself after running set_montage?

You could `logger.info` something but I think documenting makes the most sense. Maybe add something to the `Notes` of `add_reference_channels` that says it should be run before `set_montage`?

**Comment by withmywoessner** [comment]
> > Should I also add a warning that runs inside the code itself after running set_montage?
> 
> You could `logger.info` something but I think documenting makes the most sense. Maybe add something to the `Notes` of `add_reference_channels` that says it should be run before `set_montage`?

Okay, thank you @larsoner. Also, just for future reference, do you prefer all warnings to be contained in the same box? Or should each warning be in its own box like I did:
image

**Comment by larsoner** [comment]
I think separate warnings makes sense

**Comment by withmywoessner** [comment]
@larsoner Ready for merge!

File: mne/_fiff/reference.py [hunk 1]
```diff
@@ -175,6 +175,15 @@ def add_reference_channels(inst, ref_channels, copy=True):
     -------
     inst : instance of Raw | Epochs | Evoked
         Data with added EEG reference channels.
+
+    Notes
+    -----
+    .. warning::
+        When :doc:`re-referencing
+        </auto_tutorials/preprocessing/55_setting_eeg_reference>`
+        , make sure to apply the montage using :func:`mne.io.Raw.set_montage`
```
Comment (larsoner) [inline mne/_fiff/reference.py:184]:
```suggestion
        When :ref:`re-referencing <tut-set-eeg-ref>`,
        make sure to apply the montage using :func:`mne.io.Raw.set_montage`
```

File: mne/_fiff/reference.py [hunk 2]
```diff
@@ -175,6 +175,14 @@ def add_reference_channels(inst, ref_channels, copy=True):
     -------
     inst : instance of Raw | Epochs | Evoked
         Data with added EEG reference channels.
+
+    Notes
+    -----
+    .. warning::
+        When :ref:`re-referencing <tut-set-eeg-ref>`,
+        make sure to apply the montage using :func:`mne.io.Raw.set_montage`
```
Comment (larsoner) [inline mne/_fiff/reference.py:183]:
```suggestion
        make sure to apply the montage using :meth:`mne.io.Raw.set_montage`
```

**Review by larsoner — APPROVED** [review]
Just some minor link improvements, I'll mark for merge-when-green, thanks @withmywoessner !

File: mne/_fiff/meas_info.py [hunk 3]
```diff
@@ -410,6 +410,12 @@ def set_montage(
             a montage. Other channel types (e.g., MEG channels) should have
             their positions defined properly using their data reading
             functions.
+        .. warning::
+            Applying a montage will only set locations of channels that exist
+            at the time it is applied. This means when :doc:`re-referencing
+            </auto_tutorials/preprocessing/55_setting_eeg_reference>`
```
Comment (larsoner) [inline mne/_fiff/meas_info.py:416]:
```suggestion
            at the time it is applied. This means when
            :ref:`re-referencing <tut-set-eeg-ref>`
```

File: doc/changes/devel.rst [hunk 4]
```diff
@@ -45,7 +45,7 @@ Enhancements
 - Add :class:`mne.viz.ui_events.UIEvent` linking for interactive colorbars, allowing users to link figures and change the colormap and limits interactively. This supports :func:`~mne.viz.plot_evoked_topomap`, :func:`~mne.viz.plot_ica_components`, :func:`~mne.viz.plot_tfr_topomap`, :func:`~mne.viz.plot_projs_topomap`, :meth:`~mne.Evoked.plot_image`, and :meth:`~mne.Epochs.plot_image` (:gh:`12057` by `Santeri Ruuskanen`_)
 - :func:`~mne.epochs.make_metadata` now accepts ``tmin=None`` and ``tmax=None``, which will bound the time window used for metadata generation by event names (instead of a fixed time). That way, you can now for example generate metadata spanning from one cue or fixation cross to the next, even if trial durations vary throughout the recording (:gh:`12118` by `Richard Höchenberger`_)
 - Add support for passing multiple labels to :func:`mne.minimum_norm.source_induced_power` (:gh:`12026` by `Erica Peterson`_, `Eric Larson`_, and `Daniel McCloy`_ )
-
+- Added documentation to :meth:`mne.io.Raw.set_montage`and :func:`mne.add_reference_channels` to specify that montages should be set after adding reference channels (:gh:`12160` by `Jacob Woessner`_)
```
Comment (larsoner) [inline doc/changes/devel.rst:48]:
```suggestion
- Added documentation to :meth:`mne.io.Raw.set_montage` and :func:`mne.add_reference_channels` to specify that montages should be set after adding reference channels (:gh:`12160` by `Jacob Woessner`_)
```

**Pull request merged**